src/tsickle.ts

```diff
@@ -1171,60 +1156,82 @@ class Annotator extends ClosureRewriter {
+   * somewhere.
+   */
   private emitTypeAnnotationsHelper(classDecl: ts.ClassDeclaration) {
     // Gather parameter properties from the constructor, if it exists.
-    let ctors: ts.ConstructorDeclaration[] = [];
+    const ctors: ts.ConstructorDeclaration[] = [];
     let paramProps: ts.ParameterDeclaration[] = [];
-    let nonStaticProps: ts.PropertyDeclaration[] = [];
-    let staticProps: ts.PropertyDeclaration[] = [];
-    for (let member of classDecl.members) {
+    const nonStaticProps: ts.PropertyDeclaration[] = [];
+    const staticProps: ts.PropertyDeclaration[] = [];
+    const abstractMethods: ts.FunctionLikeDeclaration[] = [];
+    for (const member of classDecl.members) {
       if (member.kind === ts.SyntaxKind.Constructor) {
         ctors.push(member as ts.ConstructorDeclaration);
       } else if (member.kind === ts.SyntaxKind.PropertyDeclaration) {
-        let prop = member as ts.PropertyDeclaration;
-        let isStatic = hasModifierFlag(prop, ts.ModifierFlags.Static);
+        const prop = member as ts.PropertyDeclaration;
+        const isStatic = hasModifierFlag(prop, ts.ModifierFlags.Static);
         if (isStatic) {
           staticProps.push(prop);
         } else {
           nonStaticProps.push(prop);
         }
+      } else if (
+          hasModifierFlag(member, ts.ModifierFlags.Abstract) &&
+          (member.kind === ts.SyntaxKind.MethodDeclaration ||
+           member.kind === ts.SyntaxKind.GetAccessor ||
+           member.kind === ts.SyntaxKind.SetAccessor)) {
+        abstractMethods.push(
+            member as ts.MethodDeclaration | ts.GetAccessorDeclaration | ts.SetAccessorDeclaration);
       }
     }
 
     if (ctors.length > 0) {
-      let ctor = ctors[0];
+      const ctor = ctors[0];
       paramProps = ctor.parameters.filter(p => hasModifierFlag(p, VISIBILITY_FLAGS));
     }
 
-    if (nonStaticProps.length === 0 && paramProps.length === 0 && staticProps.length === 0) {
+    if (nonStaticProps.length === 0 && paramProps.length === 0 && staticProps.length === 0 &&
+        abstractMethods.length === 0) {
       // There are no members so we don't need to emit any type
       // annotations helper.
       return;
     }
 
     if (!classDecl.name) return;
-    let className = getIdentifierText(classDecl.name);
+    const className = getIdentifierText(classDecl.name);
 
     this.emit(`\n\nfunction ${className}_tsickle_Closure_declarations() {\n`);
     staticProps.forEach(p => this.visitProperty([className], p));
-    let memberNamespace = [className, 'prototype'];
+    const memberNamespace = [className, 'prototype'];
     nonStaticProps.forEach((p) => this.visitProperty(memberNamespace, p));
     paramProps.forEach((p) => this.visitProperty(memberNamespace, p));
+
+    for (const fnDecl of abstractMethods) {
+      const name = this.propertyName(fnDecl);
+      if (!name) {
+        this.error(fnDecl, 'anonymous abstract function');
```

Comment:
This matches previous behaviour. I think it's OK to leave it like this, as long as it doesn't break google3 I'd rather be stricter.